pyxem/signals/diffraction2d.py

```diff
@@ -466,12 +473,23 @@ def get_direct_beam_position(self, method, **kwargs):
 
         method_function = select_method_from_method_dict(method, method_dict, **kwargs)
 
+        if square_width is not None:
+            min_index = np.int(origin_coordinates[0] - square_width)
+            # fails if non-square dp
+            max_index = np.int(origin_coordinates[0] + square_width)
+            sig = self.isig[min_index:max_index, min_index:max_index]
+        else:
+            sig = self
+
         if method == 'cross_correlate':
-            shifts = self.map(method_function, inplace=False, **kwargs)
+            shifts = sig.map(method_function, inplace=False, **kwargs)
         elif method == 'blur' or method == 'interpolate':
-            centers = self.map(method_function, inplace=False, **kwargs)
+            centers = sig.map(method_function, inplace=False, **kwargs)
             shifts = origin_coordinates - centers
 
+        shifts = -1 * shifts.data
+        shifts = shifts.reshape(nav_shape_x * nav_shape_y, 2)
+
```

Comment:
This changes the return structure of this function, which seemed to be what @tinabe was using privately but it might cause issues for other parts of pyxem. So we need to verify that it doesn't or change it.